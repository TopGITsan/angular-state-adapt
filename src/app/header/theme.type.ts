// export const themes = ['os-default', 'dark', 'light'] as const;
// export type Theme = (typeof themes)[number];
export enum Theme {
  'os-default' = 'deblur',
  'dark' = 'dark_mode',
  'light' = 'light_mode',
}
export type ThemeKey = keyof typeof Theme;
export const themeValues = Object.keys(Theme).filter(
  (k): k is ThemeKey => typeof k === 'string' && k in Theme,
);
export const themeButtonIcons = Object.values(Theme).filter(
  (k): k is Theme => typeof k === 'string',
);
