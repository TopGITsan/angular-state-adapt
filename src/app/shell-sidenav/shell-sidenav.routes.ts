import { ABOUT_ROUTE } from '@about/about.route';
import { Routes } from '@angular/router';
import { WELCOME_ROUTE, welcomePath } from 'src/app/welcome/welcome.route';
import { ShellSidenvStoreService } from './shell-sidenav-store/shell-sidenav.service';
import { ShellSidenavComponent } from './shell-sidenav.component';

export const routes: Routes = [
  {
    path: '',
    component: ShellSidenavComponent,
    providers: [ShellSidenvStoreService],
    children: [
      WELCOME_ROUTE,
      ABOUT_ROUTE,
      { path: '', redirectTo: `/${welcomePath}`, pathMatch: 'full' },
      { path: '**', redirectTo: `/${welcomePath}`, pathMatch: 'full' },
    ],
  },
];
