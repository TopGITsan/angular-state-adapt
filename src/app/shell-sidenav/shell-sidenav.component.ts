import {
  closeSidenavChange$,
  toggleSidenavChange$,
} from '@actions/sidenav.actions';
import { AsyncPipe } from '@angular/common';
import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatSidenavModule } from '@angular/material/sidenav';
import { ActivatedRoute, RouterLink, RouterOutlet } from '@angular/router';
import { routeTransition } from '@shared/animations/route-transition.animation';
import { HeaderComponent } from '../header/header.component';
import { ShellSidenvStoreService } from './shell-sidenav-store/shell-sidenav.service';

@Component({
  selector: 'app-shell-sidenav',
  imports: [
    AsyncPipe,
    HeaderComponent,
    MatSidenavModule,
    RouterLink,
    RouterOutlet,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
  animations: [routeTransition],
  template: `
    @let sidenavOpened = sidenavOpened$ | async;
    @let sidenavMode = (sidenavMode$ | async) ?? 'side';

    <app-header (openSidenav)="toggleSidenav$.next($event)" />
    <mat-sidenav-container class="grow">
      <mat-sidenav
        [mode]="sidenavMode"
        [opened]="sidenavOpened"
        (closed)="closeSidenav$.next()"
        class="p-8"
      >
        <p class="m-2">
          <a [routerLink]="'home'"> home </a>
        </p>
        <p class="m-2">
          <a [routerLink]="'about'"> about </a>
        </p>
      </mat-sidenav>
      <mat-sidenav-content [@routeTransition]="route.snapshot.data">
        <router-outlet />
      </mat-sidenav-content>
    </mat-sidenav-container>
  `,
  styles: `
    :host {
      @apply flex flex-col h-full;
    }
  `,
})
export class ShellSidenavComponent {
  private readonly shellSidenavStore = inject(ShellSidenvStoreService);
  protected readonly route = inject(ActivatedRoute);
  readonly sidenavOpened$ = this.shellSidenavStore.store.isSidenavOpendState$;
  readonly sidenavMode$ = this.shellSidenavStore.store.sidenavModeState$;
  readonly toggleSidenav$ = toggleSidenavChange$;
  readonly closeSidenav$ = closeSidenavChange$;
}
