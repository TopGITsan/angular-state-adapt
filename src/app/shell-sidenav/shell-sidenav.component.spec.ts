import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ShellSidenavComponent } from './shell-sidenav.component';
import { defaultStoreProvider } from '@state-adapt/angular';
import { ActivatedRoute } from '@angular/router';
import { provideExperimentalZonelessChangeDetection } from '@angular/core';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { getTranslocoTestingModule } from '@internalization/get-transloco-testing-module.function';
import { HeaderComponent } from '../header/header.component';
import { ShellSidenvStoreService } from './shell-sidenav-store/shell-sidenav.service';

describe('ShellSidenavComponent', () => {
  let component: ShellSidenavComponent;
  let fixture: ComponentFixture<ShellSidenavComponent>;
  const fakeActivatedRoute = {
    snapshot: { data: {} },
  } as ActivatedRoute;
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [HeaderComponent, getTranslocoTestingModule()],
      providers: [
        provideExperimentalZonelessChangeDetection(),
        provideAnimationsAsync(),
        defaultStoreProvider,
        { provide: ActivatedRoute, useValue: fakeActivatedRoute },
        ShellSidenvStoreService,
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(ShellSidenavComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
});
