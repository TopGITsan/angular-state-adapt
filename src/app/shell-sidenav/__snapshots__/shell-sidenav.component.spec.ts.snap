// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShellSidenavComponent should create 1`] = `
<app-shell-sidenav
  closeSidenav$={[Function Source]}
  route={[Function Object]}
  shellSidenavStore={[Function ShellSidenvStoreService]}
  sidenavMode$={[Function Observable]}
  sidenavOpened$={[Function Observable]}
  toggleSidenav$={[Function Source]}
>
  <app-header>
    <mat-toolbar
      class="mat-toolbar mat-toolbar-single-row"
    >
      <button
        aria-label="icon-button with menu icon"
        class="mdc-icon-button mat-mdc-icon-button mat-unthemed mat-mdc-button-base"
        mat-icon-button=""
        mat-ripple-loader-centered=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
        />
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          menu
        </mat-icon>
        <span
          class="mat-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </button>
      <span>
        Less is More
      </span>
      <span
        class="flex-auto"
      />
      <a
        aria-label="GitHub Repository"
        class="mx-2 mdc-icon-button mat-mdc-icon-button mat-unthemed mat-mdc-button-base"
        href="https://github.com/TopGITsan"
        mat-icon-button=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
        target="_blank"
        title="https://github.com/TopGITsan"
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
        />
        <svg
          aria-hidden="true"
          class="w-6 h-6"
          viewBox="0 0 20 20"
        >
          <path
            d="M10 0C4.477 0 0 4.477 0 10c0 4.42 2.87 8.17 6.84 9.5.5.08.66-.23.66-.5v-1.69c-2.77.6-3.36-1.34-3.36-1.34-.46-1.16-1.11-1.47-1.11-1.47-.91-.62.07-.6.07-.6 1 .07 1.53 1.03 1.53 1.03.87 1.52 2.34 1.07 2.91.83.09-.65.35-1.09.63-1.34-2.22-.25-4.55-1.11-4.55-4.92 0-1.11.38-2 1.03-2.71-.1-.25-.45-1.29.1-2.64 0 0 .84-.27 2.75 1.02.79-.22 1.65-.33 2.5-.33.85 0 1.71.11 2.5.33 1.91-1.29 2.75-1.02 2.75-1.02.55 1.35.2 2.39.1 2.64.65.71 1.03 1.6 1.03 2.71 0 3.82-2.34 4.66-4.57 4.91.36.31.69.92.69 1.85V19c0 .27.16.59.67.5C17.14 18.16 20 14.42 20 10A10 10 0 0 0 10 0z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="mat-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </a>
      <app-theme-icon-button>
        <button
          class="mx-2 mdc-icon-button mat-mdc-icon-button"
          mat-icon-button=""
          mat-ripple-loader-centered=""
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          mat-ripple-loader-uninitialized=""
        >
          <span
            class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
          />
          <mat-icon
            aria-hidden="true"
            class="mat-icon notranslate"
            role="img"
          >
          </mat-icon>
          <span
            class="mat-focus-indicator"
          />
          <span
            class="mat-mdc-button-touch-target"
          />
        </button>
      </app-theme-icon-button>
      <app-lang-menu-button>
        <button
          aria-label="Language icon-button with a menu"
          class="mat-mdc-menu-trigger mdc-button mat-mdc-button"
          mat-button=""
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          mat-ripple-loader-uninitialized=""
        >
          <span
            class="mat-mdc-button-persistent-ripple"
          />
          <mat-icon
            aria-hidden="true"
            class="mat-icon notranslate"
            role="img"
          >
            arrow_drop_down
          </mat-icon>
          <span
            class="mdc-button__label"
          >
          </span>
          <span
            class="mat-focus-indicator"
          />
          <span
            class="mat-mdc-button-touch-target"
          />
        </button>
        <mat-menu>
        </mat-menu>
      </app-lang-menu-button>
    </mat-toolbar>
  </app-header><mat-sidenav-container
    class="mat-drawer-container mat-sidenav-container grow"
  >
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="mat-drawer mat-sidenav p-8 mat-drawer-side mat-drawer-opened"
      tabindex="-1"
    >
      <div
        cdkscrollable=""
        class="mat-drawer-inner-container"
      >
        <p
          class="m-2"
        >
          <a
            href="/"
          >
             home 
          </a>
        </p>
        <p
          class="m-2"
        >
          <a
            href="/"
          >
             about 
          </a>
        </p>
      </div>
    </mat-sidenav>
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="mat-drawer-content mat-sidenav-content"
    >
      <router-outlet />
    </mat-sidenav-content>
  </mat-sidenav-container>
</app-shell-sidenav>
`;
